packages/web-components/src/components/va-modal/va-modal.tsx

```diff
@@ -0,0 +1,336 @@
+
+    // NOTE: With this PR (https://github.com/department-of-veterans-affairs/vets-website/pull/11712)
+    // we rely on the existence of `body.modal-open` to determine if a modal is
+    // currently open and adjust programmatic scrolling if there is.
+    document.body.classList.add('modal-open');
+
+    // Conditionally track the event.
+    if (!this.disableAnalytics) {
+      const detail = {
+        componentName: 'va-modal',
+        action: 'show',
+        details: {
+          status: this.status,
+          title: this.modalTitle,
+          primaryButtonText: this.primaryButtonText,
+          secondayButtonText: this.secondaryButtonText,
+        },
+      };
+      this.componentLibraryAnalytics.emit(detail);
+    }
+  }
+
+  /**
+   * This method removes the focus trap, re-enables scrolling and
+   * removes aria-hidden="true" from external elements.
+   */
+  private teardownModal() {
+    this.focusTrap.deactivate();
+    clearAllBodyScrollLocks();
+    this.undoAriaHidden();
+
+    document.body.classList.remove('modal-open');
+  }
+
+  render() {
+    const {
+      modalTitle,
+      hideCloseButton,
+      primaryButtonClick,
+      primaryButtonText,
+      secondaryButtonClick,
+      secondaryButtonText,
+      status,
+      visible,
+    } = this;
+
+    if (!visible) return null;
+
+    const ariaLabel = `${modalTitle} modal` || 'Modal';
+    const wrapperClass = status
+      ? 'va-modal-inner va-modal-alert'
+      : 'va-modal-inner';
+    const bodyClass = status ? 'va-modal-alert-body' : 'va-modal-body';
+    const titleClass = status ? 'va-modal-alert-title' : 'va-modal-title';
+    const contentClass = status ? 'va-modal-alert-text' : undefined;
+    const ariaRole = status => {
+      if (status === 'warning' || status === 'error') {
+        return 'alertdialog';
+      }
+      return 'dialog';
+    };
+    const btnAriaLabel = modalTitle
+      ? `Close ${modalTitle} modal`
+      : 'Close modal';
+
+    return (
+      <Host
+        aria-label={ariaLabel}
+        aria-modal="true"
+        class="va-modal"
+        role={ariaRole(status)}
+      >
+        <div class={wrapperClass} tabIndex={-1}>
+          {!hideCloseButton && (
+            <div ref={el => (this.closeButtonContainer = el as HTMLDivElement)}>
+              <button
+                aria-label={btnAriaLabel}
+                class="va-modal-close"
+                onClick={e => this.handleClose(e)}
+                ref={el => (this.closeButton = el as HTMLButtonElement)}
+                type="button"
+              >
+                <i aria-hidden="true" />
+              </button>
+            </div>
+          )}
+          <div class={bodyClass}>
+            <div role="document">
+              {modalTitle && (
+                <h1 class={titleClass} tabIndex={-1}>
+                  {modalTitle}
+                </h1>
+              )}
+              <div class={contentClass}>
+                <slot></slot>
+              </div>
```

Comment:
The `contentClass` could be:

```css
.va-modal-alert-text {
  margin-top: 2rem;
  margin-bottom: 0;
}
```
I toggled the `margin-bottom` and didn't notice any difference - do we need that?

Also, could we get away with `margin-bottom: 2rem` on the `h1` just above this and not need the extra class and wrapping element?